backend/apis/clients/list-clients.api.js

```diff
@@ -1,23 +1,42 @@
-const { app, databaseError, pool, invalidRequiest } = require("../../server");
+const { app, databaseError, pool, invalidRequest } = require("../../server");
 const mysql = require("mysql");
 
 app.get("/api/clients", (req, res, next) => {
   pool.getConnection((err, connection) => {
     if (err) {
       return databaseError(req, res, err);
     }
-    var queryString = "SELECT ? AS result";
 
-    const query = mysql.format(queryString, [req.query.foo]);
+    var queryString =
+      "SELECT cl.id,cl.firstName,cl.lastName,cl.birthday,ct.zip,ct.primaryPhone,cl.addedBy as addedById,us.firstName as addedByFirstName,us.lastname as addedByLastName,cl.dateAdded FROM clients cl JOIN contactInformation ct ON cl.id = ct.clientId JOIN users us ON cl.addedBy = us.Id WHERE (cl.firstName LIKE ? AND cl.lastName LIKE ?) OR ct.zip = ? ORDER BY cl.firstName,cl.lastName DESC LIMIT 100";
+    if (req.query.page) {
+      queryString += "," + parseInt(req.query.page) * 100;
+    }
+    const getClientList = mysql.format(queryString, [
+      req.query.firstName + "%",
```

Comment:
Should we do `'%' + req.query.firstName + '%'` so that it's a "contains" instead of "startsWith"? Also, I think we should move the `%` into the query instead of into the params. The params are escaped for SQL injection, but the `%` is part of the query that we do not want escaped.

Reply:
Ill do the former. When i tried to put the % into the query the % was escaped and the query failed. Like this (cl.firstName LIKE %\'Leo\'% AND cl.lastName LIKE %\'Nieto\'%)